### PR TITLE
Switch to GH Actions & add WP.org validation

### PR DESCRIPTION
**Pull request by jazzsequence**
This pulls out the CircleCI steps for everything except Behat and adds the WP.org validation action.
Behat is omitted because there are SSH config settings that need to be dealt with.

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    - cron: "0 0 * * *"
+  push:
+    branches:
+      - '**'
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (pwtyler) [inline .github/workflows/lint-test.yml:13]: AFAIK v2 is deprecated and v3 is used below, is this copy-pasta, or intentional?

File: .github/workflows/lint-test.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          restore-keys: test-phpunit-dependencies-${{ hashFiles('composer.json') }}
+      - name: Install Composer dependencies
+        run: composer install
+      - name: Run PHPUnit
+        run: |
+          bash bin/install-wp-tests.sh wordpress_test root root 127.0.0.1 latest
+          echo "Running PHPUnit on Single Site"
+          composer phpunit
+          rm -rf $WP_TESTS_DIR $WP_CORE_DIR
+          bash bin/install-wp-tests.sh wordpress_test root root 127.0.0.1 nightly true
```
Comment (pwtyler) [inline .github/workflows/lint-test.yml:89]: 🔢I know this is copied over so optional if you want to adjust, but I believe this should be latest, not nightly?

**Review by pwtyler — DISMISSED** [review]

File: .github/workflows/lint-test.yml (hunk 2 above)
Comment (jazzsequence) [reply .github/workflows/lint-test.yml:89]: the solr-power script is doing nightly for the second test. I was unsure if that was intentional or not. I'm happy either way.

File: .github/workflows/lint-test.yml (hunk 1 above)
Comment (jazzsequence) [reply .github/workflows/lint-test.yml:13]: yep this was unintentional

**Comment by jazzsequence** [comment]
@pwtyler This still had the "dismiss stale pull requests" toggle turned on, but I actually wanted to re-request a review anyway, because as I was making changes to the PHPUnit steps, I decided to use a bash script for that stuff instead and wanted a second look.

File: bin/phpunit-test.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+bash ./bin/install-wp-tests.sh wordpress_test root root 127.0.0.1 latest
```
[the remark this answers is not in the document]
Comment (jazzsequence) [reply bin/phpunit-test.sh]: tests were passing without this and failing with. the intent of this script is not for it to be globally reusable but just to be used by GHA. Locally, you can (and should) run `composer test`, but a different configuration is needed for CI environments. I think I'm going ot pull out this change and set it back to the way it was when it was working unless you have insight into the `.: .: Is a directory` error it's getting now. https://github.com/pantheon-systems/pantheon-hud/actions/runs/5467398223/jobs/9953654931?pr=124

[the remark this answers is not in the document]
Comment (jazzsequence) [reply bin/phpunit-test.sh:5]: nevermind, I guess that wasn't actually the issue 🤦

[the remark this answers is not in the document]
Comment (jazzsequence) [reply bin/phpunit-test.sh]: fixed

```bash
SELF_DIRNAME=$(dirname "$0")
```

**Review by pwtyler — APPROVED** [review]

**Pull request merged**